arkane/encorr/bacTest.py

```diff
@@ -0,0 +1,275 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER      #
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING     #
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER         #
+# DEALINGS IN THE SOFTWARE.                                                   #
+#                                                                             #
+###############################################################################
+
+"""
+This script contains unit tests for the :mod:`arkane.encorr.bac` module.
+"""
+
+import importlib
+import os
+import tempfile
+import unittest
+from collections import Counter
+
+import numpy as np
+import pybel
+
+from rmgpy import settings
+from rmgpy.molecule import Molecule
+
+from arkane.encorr.bac import BAC
+from arkane.encorr.data import BACDataset, BOND_SYMBOLS, _pybel_to_rmg
+from arkane.encorr.reference import ReferenceDatabase
+from arkane.exceptions import BondAdditivityCorrectionError
+
+
+class TestBAC(unittest.TestCase):
+    """
+    A class for testing that the BAC class functions properly.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.model_chem_get = 'ccsd(t)-f12/cc-pvtz-f12'
+        cls.model_chem_fit = 'wb97m-v/def2-tzvpd'
+        cls.model_chem_nonexisting = 'notamethod/notabasis'
+
+        cls.bac = BAC(cls.model_chem_get)
+
+        cls.tmp_melius_params = {
+            'atom_corr': {'H': 1.0, 'C': 2.0, 'N': 3.0, 'O': 4.0, 'S': 5.0, 'F': 6.0, 'Cl': 7.0, 'Br': 8.0},
+            'bond_corr_length': {'H': 1.0, 'C': 2.0, 'N': 3.0, 'O': 4.0, 'S': 5.0, 'F': 6.0, 'Cl': 7.0, 'Br': 8.0},
+            'bond_corr_neighbor': {'H': 1.0, 'C': 2.0, 'N': 3.0, 'O': 4.0, 'S': 5.0, 'F': 6.0, 'Cl': 7.0, 'Br': 8.0},
+            'mol_corr': 1.0
+        }
+        cls.tmp_petersson_params = {'C-H': 1.0, 'C-C': 2.0, 'C=C': 3.0, 'C-O': 4.0}
+
+        # Set molecule, bonds, nums, and coords for testing Petersson and Melius BACs
+        smi = 'C=C(OSC=S)C#CC1C(=O)N=CNC1SSC(O)C#N'
+
+        mol = Molecule(smiles=smi)
+        cls.bonds = Counter(
+            f'{b.atom1.element.symbol}{BOND_SYMBOLS[b.order]}{b.atom2.element.symbol}' for b in mol.get_all_edges()
+        )
+
+        pybel_mol = pybel.readstring('smi', smi)
+        pybel_mol.addh()
+        pybel_mol.make3D()
+        cls.mol_3d = _pybel_to_rmg(pybel_mol)
+        cls.nums = [atom.number for atom in cls.mol_3d.atoms]
+        cls.coords = np.array([atom.coords for atom in cls.mol_3d.atoms])
+
+    def test_loading_parameters(self):
+        """
+        Test that BAC parameters for model chemistries are loaded
+        correctly and that errors are raised otherwise.
+        """
+        self.bac.model_chemistry = self.model_chem_get
+        self.bac.bac_type = 'p'
+        self.assertIsInstance(self.bac.bacs, dict)
+
+        self.bac.bac_type = 'm'
+        self.assertIsNone(self.bac.bacs)
```

Comment:
This will only work so long as we never retroactively fit 'ccsd(t)-f12/cc-pvtz-f12' with melius type BACs, right?

Reply:
Oh yeah. I'll change the model chemistry to the nonexistent one for this test.